runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientProduceFactory.java

```diff
@@ -2006,8 +2007,7 @@ private void encodeCrc()
                     crc.update(encodeSlotByteBuffer);
 
                     long checksum = crc.getValue();
-
-                    if (crcDataLimit == recordBatch.limit())
+                    if (crcDataLimit == recordHeaderLimit && valueCompleteSize != 0)
```

Comment:
What is the message is `null` (tombstone) instead of empty?
Perhaps this test needs to be `&& valueCompleteSize > 0` instead?

Please add an IT scenario for `null` message in addition to `empty` message to confirm whether change is necessary.